sdk/src/trace/tracer.cc

```diff
@@ -33,8 +33,20 @@ nostd::unique_ptr<trace_api::Span> Tracer::StartSpan(
     const trace_api::KeyValueIterable &attributes,
     const trace_api::StartSpanOptions &options) noexcept
 {
-  return nostd::unique_ptr<trace_api::Span>{new (std::nothrow) Span{
-      this->shared_from_this(), processor_.load(), name, attributes, options}};
+  // TODO: replace nullptr with parent context in span context
+  auto decision =
+      sampler_->ShouldSample(nullptr, trace_api::TraceId(), name, options.kind, attributes)
+          .decision;
+  if (decision == Decision::NOT_RECORD)
+  {
+    return nostd::unique_ptr<trace_api::Span>{new (std::nothrow)
```

Comment:
Could we cache a noop span instance per tracer? Is there anything else stored on the span that is needed per instance? Could potentially save a lot in allocations that way.

Reply:
Good point, but I'm not sure how to best handle this as the span is wrapped in a `nostd::unique_ptr`.

I think we should leave this to a future optimization stage, once we have benchmarks in place and can reason better about trade-offs.

Reply:
Fair, I think holding off til we have benchmarks is a very good idea. I think you could probably hold onto the span as a `shared_ptr<Span>` on the tracer and it would handle properly?